projects/plugins/jetpack/extensions/blocks/subscriptions/email-preview.js

```diff
@@ -0,0 +1,85 @@
+import apiFetch from '@wordpress/api-fetch';
+import { Button, Flex, FlexItem, Modal, TextControl } from '@wordpress/components';
+import { useSelect } from '@wordpress/data';
+import { useState } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import './email-preview.scss';
+import check from './check.svg';
+import illoShare from './illo-share.svg';
+
+export default function EmailPreview() {
+	const [ isModalOpen, setIsModalOpen ] = useState( false );
+	const [ emailSent, setEmailSent ] = useState( false );
+	const [ emailSending, setEmailSending ] = useState( false );
+	const postId = useSelect( select => select( 'core/editor' ).getCurrentPostId() );
+
+	const sendEmailPreview = () => {
+		setEmailSending( true );
+		apiFetch( {
+			path: '/wpcom/v2/send-email-preview/',
+			method: 'POST',
+			data: {
+				id: postId,
+			},
+		} )
+			.then( () => {
+				// Handle response here
+				setEmailSending( false );
+				setEmailSent( true );
+			} )
+			.catch( () => {
+				setEmailSending( false );
+
+				// Handle error here
```

Comment:
Snackbar notice would be quick to use here:

https://wordpress.github.io/gutenberg/?path=/story/components-snackbar--default

...but I think a state flag for "error" would be good, and we can toggle some message in place of success message that now appears with `emailSent`.